docs/readmes/bazel/agw_with_bazel.md

```diff
@@ -0,0 +1,544 @@
+
+In the `deps` attribute of the target that needs the external dependency the name of the external dependency repository is prefixed by `@` and followed by the Bazel target. For example as `deps = ["@aioh2_repo//:aioh2"]`.
+
+#### Pip dependencies (Python)
+
+In most cases external Python dependencies can be added as pip dependencies. For this the dependency needs to be added to the `$MAGMA_ROOT/bazel/external/requirements.in` file. If a new dependency is added, the `requirements.txt` file needs to be rebuild by
+
+```bash
+$MAGMA_ROOT/bazel/external $ pip-compile --allow-unsafe --generate-hashes --output-file=requirements.txt requirements.in
+```
+
+> Info: See the `bazel/external/requirements_README.md` for more information regarding the `requirements.in` file.
+
+The pip dependency, e.g. `grpcio`, can now be added to a target by adding a requirement statement `deps = [requirement("grpcio")]`.
+
+### Cleaning
+
+The Bazel output, i.e. the artifacts, can be cleaned by running [`bazel clean`](https://bazel.build/docs/user-manual#cleaning-build-outputs), which will remove all outputs from all previous builds for all configurations. The entire working tree created by Bazel can be removed by running `bazel clean --expunge`.
+
+Usually these commands should not be needed, however they can be used to reclaim disk space or to recover a consistent state when debugging.
+
+> Info: The `bazel clean` command does **not** remove the disk caches. To remove the disk caches the cache folders, e.g. `$MAGMA_ROOT/.bazel-cache` and `$MAGMA_ROOT/.bazel-cache-repo`, need to be removed manually. The exact locations of the disk caches may depend on the environment.
+
+## Testing
+
+There are four categories of tests that concern the AGW, each of these categories is covered in the following sections:
+
+- [Unit tests](#unit-tests)
+- [LTE integration tests](#lte-integration-tests)
+- [Python sudo tests](#python-sudo-tests)
+- [Load tests](#load-tests)
+
+> Warning: Note that not all Bazel test targets in the Magma AGW can be executed directly with Bazel. These special targets are tagged in their attributes and the following sections explain how they can be run.
+
+### Unit tests
+
+See the [notes on unit testing](../lte/dev_unit_testing.md) for more detailed information and examples on unit testing.
+
+The unit tests can be run on any of the Docker [environments](#environments) or on the magma-dev VM.
+
+To run unit tests with Bazel, use the command `bazel test` and then specify the [Bazel targets](#targets) using the same [command line syntax](#target-command-line-syntax) described in the build section.
+
+For example, to run all unit tests, that are not tagged as manual, run `bazel test //...` and to run a specific unit test specify the test target directly, e.g. `bazel test //lte/gateway/c/core/oai/test/pipelined_client:pipelined_client_test`.
```

Comment:
I cannot reproduce this error - this is a cpp unit test that is also running in CI. If you post your logging we could have a look if there is something wrong with your local setup?

Reply:
Please find the snapshot of failed log,bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/__main__/lte/gateway/python/magma/pipelined/tests/app/start_pipelined.py:179:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def assert_pipelined_not_running():
        """
        As Ryu applications shoudn't be started if the magma@pipelined service is
        running we need to verify if pipelined is active. If service is running
        throws a ServiceRunningError exception.

        This can be done using the command:
            systemctl is-active magma@pipelined
        If service is pipelined, this returns an error code 3 & message "inactive"
        """
        try:
            output = subprocess.check_output(
                ["systemctl", "is-active", "magma@pipelined"],
            )
        except subprocess.CalledProcessError as e:
            if "inactive" not in str(e.output, 'utf-8'):
                raise ServiceRunningError(
                    "Pipelined is running, 'systemctl is-active magma@pipelined'" +
                    "caused an error code %d, exception - %s"
                    % (e.returncode, str(e.output, 'utf-8').strip()),
                )
        else:
>           raise ServiceRunningError(
                "Pipelined is running, 'systemctl is-active magma@pipelined'" +
                "output - %s" % str(output, 'utf-8').strip(),
            )
E           magma.pipelined.tests.app.exceptions.ServiceRunningError: Pipelined is running, 'systemctl is-active magma@pipelined'output - active

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/__main__/lte/gateway/python/magma/pipelined/tests/app/start_pipelined.py:159: ServiceRunningError

During handling of the above exception, another exception occurred:

cls = <class 'test_vlan_learn.VlanLearnTest'>

    @classmethod
    def setUpClass(cls):
        """
        Starts the thread which launches ryu apps

        Create a testing bridge, add a port, setup the port interfaces. Then
        launch the ryu apps for testing pipelined. Gets the references
        to apps launched by using futures.
        """
        super(VlanLearnTest, cls).setUpClass()
        warnings.simplefilter('ignore')
        cls.service_manager = create_service_manager(
            [],
            ['ue_mac', 'vlan_learn'],
        )

        vlan_learn_controller_reference = Future()
        testing_controller_reference = Future()
        test_setup = TestSetup(
            apps=[
                PipelinedController.VlanLearn,
                PipelinedController.Testing,
                PipelinedController.StartupFlows,
            ],
            references={
                PipelinedController.VlanLearn:
                    vlan_learn_controller_reference,
                PipelinedController.Testing:
                    testing_controller_reference,
                PipelinedController.StartupFlows:
                    Future(),
            },
            config={
                'bridge_name': cls.BRIDGE,
                'bridge_ip_address': cls.BRIDGE_IP,
                'internal_ip_subnet': '192.168.0.0/16',
                'ovs_gtp_port_number': 32768,
                'clean_restart': True,
            },
            mconfig=None,
            loop=None,
            service_manager=cls.service_manager,
            integ_test=False,
        )

        BridgeTools.create_bridge(cls.BRIDGE, cls.IFACE)

>       cls.thread = start_ryu_app_thread(test_setup)

lte/gateway/python/magma/pipelined/tests/test_vlan_learn.py:85:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/__main__/lte/gateway/python/magma/pipelined/tests/pipelined_test_util.py:184: in start_ryu_app_thread
    exit()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Use exit() or Ctrl-D (i.e. EOF) to exit, code = None

    def __call__(self, code=None):
        # Shells like IDLE catch the SystemExit, but listen when their
        # stdin wrapper is closed.
        try:
            sys.stdin.close()
        except:
            pass
>       raise SystemExit(code)
E       SystemExit: None

/usr/lib/python3.8/_sitebuiltins.py:26: SystemExit
----------------------------------------------------- Captured log setup -----------------------------------------------------
ERROR    root:pipelined_test_util.py:183 Failed to start test apps in separate thread: Pipelined is running, 'systemctl is-active magma@pipelined'output - active
====================================================== warnings summary ======================================================
bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__prometheus_client/prometheus_client/decorator.py:81
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__prometheus_client/prometheus_client/decorator.py:81: DeprecationWarning: invalid escape sequence \s
    DEF = re.compile('\s*def\s*([_\w][_\w\d]*)\s*\(')

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__eventlet/eventlet/patcher.py:1
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__eventlet/eventlet/patcher.py:1: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    import imp

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__dnspython/dns/hash.py:23
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__dnspython/dns/hash.py:23: DeprecationWarning: dns.hash module will be removed in future versions. Please use hashlib instead.
    warnings.warn(

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__dnspython/dns/namedict.py:35
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__dnspython/dns/namedict.py:35: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    class NameDict(collections.MutableMapping):

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__dnspython/dns/zone.py:737
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__dnspython/dns/zone.py:737: DeprecationWarning: invalid escape sequence \$
    is_generate1 = re.compile("^.*\$({(\+|-?)(\d+),(\d+),(.)}).*$")

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__dnspython/dns/zone.py:738
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__dnspython/dns/zone.py:738: DeprecationWarning: invalid escape sequence \$
    is_generate2 = re.compile("^.*\$({(\+|-?)(\d+)}).*$")

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__dnspython/dns/zone.py:739
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__dnspython/dns/zone.py:739: DeprecationWarning: invalid escape sequence \$
    is_generate3 = re.compile("^.*\$({(\+|-?)(\d+),(\d+)}).*$")

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/ryu_repo/ryu/lib/packet/cfm.py:271
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/ryu_repo/ryu/lib/packet/cfm.py:271: SyntaxWarning: "is not" with a literal. Did you mean "!="?
    assert interval is not 0

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/__main__/lte/gateway/python/magma/pipelined/ifaces.py:22
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/__main__/lte/gateway/python/magma/pipelined/ifaces.py:22: DeprecationWarning: "@coroutine" decorator is deprecated since Python 3.8, use "async def" instead
    def monitor_ifaces(iface_names):

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__pymemoize/memoize/options.py:2
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/python_deps_pypi__pymemoize/memoize/options.py:2: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.10 it will stop working
    from collections import Callable

bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/bcc_repo/bcc/__init__.py:745
  /home/vagrant/magma/bazel-bin/lte/gateway/python/magma/pipelined/tests/test_vlan_learn.runfiles/bcc_repo/bcc/__init__.py:745: DeprecationWarning: invalid escape sequence \.
    elif re.match(b'^.*\.cold(\.\d+)?$', fn):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
------------------- generated xml file: /var/tmp/test_results/sudotest_reports/test_vlan_learn_report.xml --------------------
================================================== short test summary info ===================================================
ERROR lte/gateway/python/magma/pipelined/tests/test_vlan_learn.py::VlanLearnTest::testFlowSnapshotMatch - SystemExit: None
=============================================== 11 warnings, 1 error in 4.02s ================================================
INFO:root:Final report is generated at: /var/tmp/test_results/sudotests_report.xml
SUMMARY: 5/41 tests were successful.
  //lte/gateway/python/magma/pipelined/tests:test_classifier_mme_flow_dl: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_classifier_traffic: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_classifier: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_ue_passthrough: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_ue_mac: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_qos: PASSED
  //lte/gateway/python/magma/pipelined/tests:test_arp: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_arp_non_nat: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_gy: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_dpi: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_check_quota: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_ng_servicer_session: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_ebpf_ul_dp: PASSED
  //lte/gateway/python/magma/pipelined/tests:test_non_nat: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_ng_servicer_node: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_enforcement_5g: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_inout_non_nat: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_redirect: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_tunnel_learn: FAILED
  //lte/gateway/python/magma/mobilityd/tests:test_ip_allocator_dhcp_with_vlan: FAILED
  //lte/gateway/python/magma/pipelined/tests:ng_node_rpc_servicer: PASSED
  //lte/gateway/python/magma/pipelined/tests:test_li_mirror: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_vlan_learn: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_he: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_inout: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_access_control: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_ipv6_solicitation: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_internal_pkt_ipfix_export: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_ebpf_dl_dp: PASSED
  //lte/gateway/python/magma/pipelined/tests:test_restart_resilience: FAILED
  //lte/gateway/python/magma/mobilityd/tests:test_ip_allocator_dhcp_e2e: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_cwf_restart_resilience: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_paging: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_enforcement_stats: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_uplink_bridge: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_enforcement: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_qfi_gtp: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_conntrack: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_qos_pyroute2: PASSED
  //lte/gateway/python/magma/pipelined/tests:test_pull_stats: FAILED
  //lte/gateway/python/magma/pipelined/tests:test_ipv6_solicitation_5g: FAILED
vagrant@magma-dev:~/magma$

Reply:
Which command did you run?

Reply:
These pipelined tests fail because they cannot be run directly with Bazel - see section `Python sudo tests` on how to run the sudo tests.
However if you run `bazel test //...` they will never be considered for unit testing because they are tagged `    tags = TAG_SUDO_TEST,` which in particular includes the tag `TAG_MANUAL`. This prevents Bazel from ever running them with `bazel test`, unless you specifically request one of them.

Reply:
Just to be sure I tested both commands, mentioned above
- `bazel test //...`
- `bazel test //lte/gateway/c/core/oai/test/pipelined_client:pipelined_client_test`

 locally on master and they both return okay.

Reply:
I had executed "bazel test //lte/gateway/c/core/oai/test/pipelined_client:pipelined_client_test"

Reply:
On execution of "bazel test //..." getting below error
vagrant@magma-dev:~/magma$ bazel test //...
INFO: Invocation ID: 324bddfc-c174-400c-82aa-15e1a3414019
INFO: Reading 'startup' options from /home/vagrant/magma/.bazelrc: --output_base=/var/tmp/bazel, --host_jvm_args=-Xmx8g
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=126
INFO: Reading rc options for 'test' from /home/vagrant/magma/.bazelrc:
  Inherited 'common' options: --experimental_allow_tags_propagation --repository_cache=/var/cache/bazel-cache-repo
INFO: Reading rc options for 'test' from /home/vagrant/magma/.bazelrc:
  Inherited 'build' options: --announce_rc --color=yes --cxxopt=-std=c++14 --strip=never --per_file_copt=^lte/gateway/c/.*$@-g --per_file_copt=^external/.*$@-Wno-all,-Wno-format-security,-Wno-deprecated-declarations --disk_cache=/var/cache/bazel-cache --test_env=PATH=/bin:/usr/bin:/usr/local/bin:/usr/sbin --test_env=MAGMA_ROOT --test_env=S1AP_TESTER_ROOT --test_env=GO_TEST_WRAP_TESTV=1 --per_file_copt=^lte/gateway/c/core/.*$@-DPACKAGE_BUGREPORT="TBD" --per_file_copt=^lte/gateway/c/core/.*$@-DPACKAGE_VERSION="0.1" --per_file_copt=^lte/gateway/c/core/oai/tasks/amf/.*$@-DPACKAGE_NAME="AMF" --per_file_copt=^lte/gateway/c/core/oai/tasks/mme_app/.*$@-DPACKAGE_NAME="MME" --per_file_copt=^lte/gateway/c/core/oai/tasks/sgw/.*$@-DPACKAGE_NAME="S/P-GW" --javacopt=-source 8 --javacopt=-target 8 --java_runtime_version=remotejdk_11
INFO: Reading rc options for 'test' from /home/vagrant/magma/bazel/bazelrcs/vm.bazelrc:
  Inherited 'build' options: --config=vm
INFO: Reading rc options for 'test' from /home/vagrant/magma/.bazelrc:
  'test' options: --test_output=errors --per_file_copt=^lte/gateway/c/core/.*$@-DMME_UNIT_TEST
INFO: Found applicable config definition build:vm in file /home/vagrant/magma/.bazelrc: --config=specify_vm_cc
INFO: Found applicable config definition build:specify_vm_cc in file /home/vagrant/magma/.bazelrc: --action_env=CC=/usr/bin/gcc
ERROR: /home/vagrant/magma/temp/BUILD.bazel:16:8: no such target '//lte/gateway/c/core:core': target 'core' not declared in package 'lte/gateway/c/core' defined by /home/vagrant/magma/lte/gateway/c/core/BUILD.bazel and referenced by '//temp:lib_mme_app_map_test'
ERROR: Analysis of target '//temp:lib_mme_app_map_test' failed; build aborted: Analysis failed
INFO: Elapsed time: 1.493s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)
FAILED: Build did NOT complete successfully (0 packages loaded, 0 targets configured)

Reply:
Am I missing something here?

Reply:
I just ran `bazel test //...` from the current `magma-dev` VM and it works with the code from master.

One thing I find surprising about the logging that you posted is that the options from the `.bazelrc` seem to be missing `*` symbols in the regex expressions. E.g. `--per_file_copt=^lte/gateway/c/.$@-g` In the code on master this is ` --per_file_copt=^lte/gateway/c/.*$@-g`. 

Also the lines
```
ERROR: /home/vagrant/magma/temp/BUILD.bazel:16:8: no such target '//lte/gateway/c/core:core': target 'core' not declared in package 'lte/gateway/c/core' defined by /home/vagrant/magma/lte/gateway/c/core/BUILD.bazel and referenced by '//temp:lib_mme_app_map_test'
```
sound like there is a folder `home/vagrant/magma/temp` in your repository which contains a `BUILD` file that is not checked in. 

Are there some changes to the code? Could you check out the current master and try again?

Could you please post the output of `bazel info` with the same code state and environment?

For me this outputs:
```
vagrant@magma-dev:~/magma$ bazel info
INFO: Invocation ID: 9877abaa-71bb-4ecc-a9f4-6772298ada03
INFO: Reading 'startup' options from /home/vagrant/magma/.bazelrc: --output_base=/var/tmp/bazel, --host_jvm_args=-Xmx8g
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=207
INFO: Reading rc options for 'info' from /home/vagrant/magma/.bazelrc:
  Inherited 'common' options: --experimental_allow_tags_propagation --repository_cache=/var/cache/bazel-cache-repo
INFO: Reading rc options for 'info' from /home/vagrant/magma/.bazelrc:
  Inherited 'build' options: --announce_rc --color=yes --cxxopt=-std=c++14 --strip=never --per_file_copt=^lte/gateway/c/.*$@-g --per_file_copt=^external/.*$@-Wno-all,-Wno-format-security,-Wno-deprecated-declarations --disk_cache=/var/cache/bazel-cache --test_env=PATH=/bin:/usr/bin:/usr/local/bin:/usr/sbin --test_env=MAGMA_ROOT --test_env=S1AP_TESTER_ROOT --test_env=GO_TEST_WRAP_TESTV=1 --per_file_copt=^lte/gateway/c/core/.*$@-DPACKAGE_BUGREPORT="TBD" --per_file_copt=^lte/gateway/c/core/.*$@-DPACKAGE_VERSION="0.1" --per_file_copt=^lte/gateway/c/core/oai/tasks/amf/.*$@-DPACKAGE_NAME="AMF" --per_file_copt=^lte/gateway/c/core/oai/tasks/mme_app/.*$@-DPACKAGE_NAME="MME" --per_file_copt=^lte/gateway/c/core/oai/tasks/sgw/.*$@-DPACKAGE_NAME="S/P-GW" --javacopt=-source 8 --javacopt=-target 8 --java_runtime_version=remotejdk_11
INFO: Reading rc options for 'info' from /home/vagrant/magma/bazel/bazelrcs/vm.bazelrc:
  Inherited 'build' options: --config=vm
INFO: Found applicable config definition build:vm in file /home/vagrant/magma/.bazelrc: --config=specify_vm_cc
INFO: Found applicable config definition build:specify_vm_cc in file /home/vagrant/magma/.bazelrc: --action_env=CC=/usr/bin/gcc
bazel-bin: /var/tmp/bazel/execroot/__main__/bazel-out/k8-fastbuild/bin
bazel-genfiles: /var/tmp/bazel/execroot/__main__/bazel-out/k8-fastbuild/bin
bazel-testlogs: /var/tmp/bazel/execroot/__main__/bazel-out/k8-fastbuild/testlogs
character-encoding: file.encoding = ISO-8859-1, defaultCharset = ISO-8859-1
command_log: /var/tmp/bazel/command.log
committed-heap-size: 689MB
execution_root: /var/tmp/bazel/execroot/__main__
gc-count: 188
gc-time: 5212ms
install_base: /home/vagrant/.cache/bazel/_bazel_vagrant/install/41b71f1bb3ce13f20cfeeb31a9357113
java-home: /home/vagrant/.cache/bazel/_bazel_vagrant/install/41b71f1bb3ce13f20cfeeb31a9357113/embedded_tools/jdk
java-runtime: OpenJDK Runtime Environment (build 11.0.6+10-LTS) by Azul Systems, Inc.
java-vm: OpenJDK 64-Bit Server VM (build 11.0.6+10-LTS, mixed mode) by Azul Systems, Inc.
max-heap-size: 8589MB
output_base: /var/tmp/bazel
output_path: /var/tmp/bazel/execroot/__main__/bazel-out
package_path: %workspace%
release: release 5.2.0
repository_cache: /var/cache/bazel-cache-repo
server_log: /var/tmp/bazel/java.log.magma-dev.vagrant.log.java.20230113-072254.3375
server_pid: 3375
used-heap-size: 430MB
workspace: /home/vagrant/magma
```


And thank you again for taking such a close look at the documentation!

Reply:
Discussed on Zoom - solved.